src/pocketmine/network/mcpe/protocol/types/LegacySkinAdapter.php

```diff
@@ -43,11 +43,16 @@ public function toSkinData(Skin $skin) : SkinData{
 
 	public function fromSkinData(SkinData $data) : Skin{
 		$capeData = $data->getCapeImage()->getData();
+		$geometryName = "";
+		$resourcePatch = json_decode($data->getResourcePatch(), true);
+		if(isset($resourcePatch["geometry"]) && isset($resourcePatch["geometry"]["default"])){
```

Comment:
This isn't sufficient. The `geometry` key might not be an array, or the `default` key might not be a string.

Reply:
In addition, the player should be rejected with `Invalid skin` if the resource patch is invalid.